src/components/Input/AmountInput.tsx

```diff
@@ -1,26 +1,35 @@
 import { Input, InputAdornment } from '@material-ui/core'
-import React, { CSSProperties } from 'react'
+import React, { CSSProperties, useState } from 'react'
 import useStyles from './style'
 import classNames from 'classnames'
 
 interface IProps {
-  setValue: (value: string) => void
   currency: string
-  value?: string
+  initValue?: string
   error?: string | null
   className?: string
   style?: CSSProperties
 }
 
+type inputString = { target: { value: string } }
+
 export const AmountInput: React.FC<IProps> = ({
-  setValue,
   currency,
-  value,
+  initValue = '',
   error,
   className,
   style
 }) => {
   const classes = useStyles()
+  const [value, setValue] = useState(initValue)
```

Comment:
This is problematic of having this inside. How will you set max value ? or use this value with redux ?